MatrixSDK/Threads/MXThread.swift

```diff
@@ -49,14 +56,39 @@ public class MXThread: NSObject {
         super.init()
     }
     
-    internal func addEvent(_ event: MXEvent) {
-        guard eventsMap[event.eventId] == nil else {
-            //  do not re-add the event
-            return
-        }
+    /// Add an event to the thread
+    /// - Parameter event: Event
+    /// - Returns: true if handled, false otherwise
+    @discardableResult
+    internal func addEvent(_ event: MXEvent) -> Bool {
         eventsMap[event.eventId] = event
+        updateNotificationsCount()
+        if event.sender == session?.myUserId {
```

Comment:
If `event.sender` is nil and `session` is nil? It will be marked as read. Sure we are missing nullability on ObjC :/

Reply:
I've added checks for that.